docs: Fix broken permalink icon

1109lun opened the pull request:
closes: #46170

After the Sphinx upgrade, the permalink icon was rendering incorrectly as text artifacts (e.g., "[link]" or "¶") instead of the intended icon.

This PR fixes the issue by:
1.  **`conf.py`**: Explicitly setting `html_permalinks_icon` to a clean, solid SVG chain icon.
2.  **`custom.css`**:
    * Removing the legacy `::after { content: " [link]" }` style that caused the text artifact.
    * Forcing the icon visibility (removing previous hidden states).
    * Setting the icon color to Airflow Blue (`#017cee`).
    * Using `inline-flex` to ensure perfect vertical alignment with headings.

1109lun committed:
docs: Fix broken permalink icon

After the Sphinx upgrade, the permalink icon was rendering
incorrectly as text artifacts (for example, "[link]").

This change explicitly sets a custom SVG icon in conf.py and
updates custom.css to enforce visibility, correct the color to
Airflow blue, and ensure proper vertical alignment.

choo121600 reviewed (commented):
It looks like some unnecessary changes. Please revert this part.

Lee-W commented on devel-common/sphinx_design/static/custom.css:
```diff
@@ -7,7 +7,7 @@
  * with the License.  You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
```
```suggestion
 *   http://www.apache.org/licenses/LICENSE-2.0
```

Lee-W reviewed (commented):
I feel we'll have some static check failures. Please take care of them. Also, let's have the before-and-after screenshot. Thanks

1109lun committed:
docs: Revert accidental whitespace in custom.css

Restore original spacing in license header line that was changed unintentionally.

1109lun committed:
fix: Add missing newline at end of file in custom.css and conf.py

1109lun committed:
style: Fix CI formatting errors with ruff

Lee-W commented on devel-common/sphinx_design/static/custom.css:
```diff
@@ -57,14 +57,30 @@ section ol li p:last-child, section ul li p:last-child {
   margin-bottom: 0 !important;
 }
 
+/* === Fix for broken permalink icon (Issue #46170) === */
```
```suggestion
/* Fix for broken permalink icon */
```

Lee-W replied:
We can use `git blame` to find the root PR. I think this we can skip this information here

1109lun replied:
Thanks for the reminder!

Lee-W reviewed (approved):
Mostly good. One nitpick

1109lun committed:
docs: Change the comment for broken permalink icon in custom.css

jscheffl reviewed (commented):
Cool!
